CIP-interoperable-programmable-tokens/README.md

```diff
@@ -0,0 +1,172 @@
+Category: Plutus
+Status: Proposed
+Authors:
+    - Philip DiSarro <philipdisarro@gmail.com>
+Implementors: []
+Discussions:
+    - https://github.com/cardano-foundation/CIPs/pull/444
+Created: 2024-12-3
+License: Apache-2.0
+---
+
+<!-- Existing categories:
+
+- Meta     | For meta-CIPs which typically serves another category or group of categories.
+- Wallets  | For standardisation across wallets (hardware, full-node or light).
+- Tokens   | About tokens (fungible or non-fungible) and minting policies in general.
+- Metadata | For proposals around metadata (on-chain or off-chain).
+- Tools    | A broad category for ecosystem tools not falling into any other category.
+- Plutus   | Changes or additions to Plutus
+- Ledger   | For proposals regarding the Cardano ledger (including Reward Sharing Schemes)
+- Catalyst | For proposals affecting Project Catalyst / the Jörmungandr project
+
+-->
+
+## Abstract
+
+This CIP proposes a robust framework for the issuance of interoperable programmable tokens on Cardano. Unlike all its predecessors, this framework allows these tokens to be used in existing dApps, and does not require dApps to be developed specifically for these tokens. 
+
+## Motivation: why is this CIP necessary?
+
+This CIP proposes a solution to Cardano Problem Statement 3 ([CPS-0003](https://github.com/cardano-foundation/CIPs/pull/382/files?short_path=5a0dba0#diff-5a0dba075d998658d72169818d839f4c63cf105e4d6c3fe81e46b20d5fd3dc8f)).
+
+With this framework we achieve programmability over the transfer of tokens (meta-tokens) and their lifecycle without sacrificing composability with existing dApps. 
+
+Answers to the open questions of CPS-0003:
+
+1) How to support wallets to start supporting validators?
+
+For every user address you can easily derive the equivalent smart token address. So to obtain a users total wallet balance including their programmable tokens, the wallet can query their programmable token address and their normal address and combine the two results.  
+
+2) How would wallets know how to interact with these tokens? - smart contract registry?
+
+For any given programmable token transfer, wallets can easily determine which stake script needs to be invoked (that contains the transfer logic) directly from the chain (no offchain registry required) by querying the original minting tx. 
+
+3) Is there a possibility to have a transition period, so users won't have their funds blocked until the wallets start supporting smart tokens?
+
+This framework will not cause any funds to be blocked. Transfers of non-programmable tokens will remain unaffected.
+
+4) Can this be achieved without a hard fork?
+
+Yes, this framework has been possible since the Chang hard fork. 
+
+5) How to make validator scripts generic enough without impacting costs significantly?
+
+The impact that the required script executions have on the cost of transactions is negligible.  
+
+
+## Specification
+
+# Smart Token Directory & Minting policy
```

Comment:
```suggestion
### Smart Token Directory & Minting policy
```
These are all part of the specification, so need to be subheadings of it; etc. for following sections...